gen/gen.go

```diff
@@ -0,0 +1,218 @@
+			"CallbackArgs": CallbackArgs,
+		})
+	t, err := t.Parse(tmpl)
+	if err != nil {
+		return fmt.Errorf("GenCHeader: failed to parse template: %w", err)
+	}
+	if err := t.Execute(dst, data); err != nil {
+		return fmt.Errorf("GenCHeader: failed to execute template: %w", err)
+	}
+	return nil
+}
+
+func CValue(s string) string {
+	switch s {
+	case "usize_max":
+		return "SIZE_MAX"
+	case "uint32_max":
+		return "0xffffffffUL"
+	case "uint64_max":
+		return "0xffffffffffffffffULL"
+	default:
+		var num string
+		var base int
+		if strings.HasPrefix(s, "0x") {
+			base = 16
+			num = strings.TrimPrefix(s, "0x")
+		} else {
+			base = 10
+			num = s
+		}
+		v, err := strconv.ParseUint(num, base, 64)
+		if err != nil {
+			panic(fmt.Errorf("CValue: failed to parse \"%s\": %w", s, err))
+		}
+		var suffix string
+		if v <= math.MaxUint32 {
+			suffix = "UL"
+		} else {
+			suffix = "ULL"
+		}
+		return "0x" + strconv.FormatUint(v, 16) + suffix
+	}
+}
+
+func CType(typ string, pointerType PointerType) string {
+	appendModifiers := func(s string, pointerType PointerType) string {
+		var sb strings.Builder
+		sb.WriteString(s)
+		switch pointerType {
+		case PointerTypeImmutable:
+			sb.WriteString(" const *")
+		case PointerTypeMutable:
+			sb.WriteString(" *")
+		}
+		return sb.String()
+	}
+	switch typ {
+	case "bool":
+		return appendModifiers("WGPUBool", pointerType)
+	case "string":
+		return appendModifiers("char", PointerTypeImmutable)
+	case "uint16":
+		return appendModifiers("uint16_t", pointerType)
+	case "uint32":
+		return appendModifiers("uint32_t", pointerType)
+	case "uint64":
+		return appendModifiers("uint64_t", pointerType)
+	case "usize":
+		return appendModifiers("size_t", pointerType)
+	case "int16":
+		return appendModifiers("int16_t", pointerType)
+	case "int32":
+		return appendModifiers("int32_t", pointerType)
+	case "float32":
+		return appendModifiers("float", pointerType)
+	case "float64":
+		return appendModifiers("double", pointerType)
+	case "c_void":
+		return appendModifiers("void", pointerType)
+	}
+
+	switch {
+	case strings.HasPrefix(typ, "enum."):
+		ctype := "WGPU" + PascalCase(strings.TrimPrefix(typ, "enum."))
+		return appendModifiers(ctype, pointerType)
+	case strings.HasPrefix(typ, "bitflag."):
+		ctype := "WGPU" + PascalCase(strings.TrimPrefix(typ, "bitflag.")) + "Flags"
+		return appendModifiers(ctype, pointerType)
+	case strings.HasPrefix(typ, "struct."):
+		ctype := "WGPU" + PascalCase(strings.TrimPrefix(typ, "struct."))
+		return appendModifiers(ctype, pointerType)
+	case strings.HasPrefix(typ, "function_type."):
+		ctype := "WGPU" + PascalCase(strings.TrimPrefix(typ, "function_type."))
+		return appendModifiers(ctype, pointerType)
+	case strings.HasPrefix(typ, "object."):
+		ctype := "WGPU" + PascalCase(strings.TrimPrefix(typ, "object."))
+		return appendModifiers(ctype, pointerType)
+	default:
+		return ""
+	}
+}
+
+func FunctionArgs(f Function, o *Object) string {
+	var sb strings.Builder
```

Comment:
```suggestion
	sb := &strings.Builder{}
```

Then you can drop all the `&` on the `Fprintf` calls.